app/uk/gov/hmrc/gform/sharedmodel/formtemplate/DateValue.scala

```diff
@@ -25,6 +27,24 @@ final case class ExactDateValue(year: Int, month: Int, day: Int) extends DateVal
 final case class NextDateValue(month: Int, day: Int) extends DateValue
 final case class PreviousDateValue(month: Int, day: Int) extends DateValue
 
+final case class FirstDayValue(year: String, month: String) extends DateValue {
+  val day = 1
+
+}
+final case class LastDayValue(year: String, month: String) extends DateValue {
+
+  private val isLeapYear = (year: Int) =>
```

Comment:
`private val isLeapYear` function is defined two times in the code base. We need to have only one implementation to keep our code base maintainable.

Reply:
... and doesn't you having to work it out for yourself when using an elaborate temporal library not ring alarm bells.  I believe you can use java.time.Year#isLeap(long)